frontend/packages/shared/src/hooks/mutations/useUpdateOptionListMutation.ts

```diff
@@ -0,0 +1,56 @@
+import type { MutationMeta } from '@tanstack/react-query';
+import { QueryKey } from 'app-shared/types/QueryKey';
+import type { Option } from 'app-shared/types/Option';
+import { useQueryClient, useMutation } from '@tanstack/react-query';
+import { useServicesContext } from 'app-shared/contexts/ServicesContext';
+import type { OptionsLists } from 'app-shared/types/api/OptionsLists';
+
+export interface UpdateOptionListMutationArgs {
+  optionListId: string;
+  optionsList: Option[];
+}
+
+export const useUpdateOptionListMutation = (org: string, app: string, meta?: MutationMeta) => {
+  const queryClient = useQueryClient();
+  const { updateOptionList } = useServicesContext();
+
+  return useMutation({
+    mutationFn: ({ optionListId, optionsList }: UpdateOptionListMutationArgs) => {
+      return updateOptionList(org, app, optionListId, optionsList);
+    },
+    onSuccess: (updatedOptionsList: OptionsLists) => {
+      const convertedNewData = convertToMap(updatedOptionsList);
+      const keys = [...convertedNewData.keys()];
+      if (keys.length > 1) {
+        const optionListId = keys[0];
+        const convertedUpdatedOptionsList = convertedNewData.get(optionListId);
+
+        const oldData: Map<string, Option[]> = queryClient.getQueryData([
+          QueryKey.OptionLists,
+          org,
+          app,
+        ]);
+        oldData.set(optionListId, convertedUpdatedOptionsList);
+
+        queryClient.setQueryData([QueryKey.OptionLists, org, app], (): Map<string, Option[]> => {
+          return oldData;
+        });
+      }
+    },
+    meta,
+  });
+};
```

Comment:
Here is a suggestion that avoids conversion and where backend only passes the updated option list 
```suggestion
export const useUpdateOptionListMutation = (org: string, app: string, meta?: MutationMeta) => {
  const queryClient = useQueryClient();
  const { updateOptionList } = useServicesContext();

  return useMutation<Option[], Error, UpdateOptionListMutationArgs>({
    mutationFn: ({ optionListId, optionsList }: UpdateOptionListMutationArgs) => {
      return updateOptionList(org, app, optionListId, optionsList);
    },
    onSuccess: (updatedOptionList: Option[], { optionListId }) => {
      const oldData: Record<string, Option[]> = queryClient.getQueryData([
          QueryKey.OptionLists,
          org,
          app,
        ]);
      oldData[optionListId] = updatedOptionList;

        queryClient.setQueryData([QueryKey.OptionLists, org, app], oldData);
    },
    meta,
  });
};
```